Sources/NIO/BaseSocketChannel.swift

```diff
@@ -921,7 +921,12 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
                 } else {
                     // we don't have a socket error, this must be connection reset without an error then
                     // this path should only be executed on Linux (EPOLLHUP, no EPOLLERR)
-                    error = IOError(errnoCode: ECONNRESET, reason: "connection reset (no error set)")
+                    #if os(Linux)
+                    let message: String = "connection reset (no error set)"
+                    #else
+                    let message: String = "BUG IN SwiftNIO (possibly #572), please report! Connection reset (no error set)."
```

Comment:
Should this `precondition` instead?

Reply:
@Lukasa I was on the fence but it doesn't actually cause any bad behaviour at all. It's "just" potentially not quite the error message in that case.

Reply:
Heh, I suspect that even if this path does execute on Darwin no-one will *ever* report a bug with it. But it doesn't hurt.